CHANGELOG.md

```diff
@@ -10,12 +10,289 @@ New features:
+
+In order to forbid such unsafe coercion we added a _nominal_ annotation to the first parameter of `Map`:
+
+```purescript
+type role Map nominal representational
+```
+
+Annotated roles are compared against the roles inferred by the compiler so it is not possible to compromise safety by ascribing too permissive roles, except for foreign types.
+
+### Other changes
+
+#### Breaking
+
+* Add compiler support for `Coercible` based `Newtype` (#3975, @fsoikin)
+
+We added a `Coercible` superclass to `Data.Newtype.Newtype` in order to implement `unwrap`, `wrap` and most newtype combinators with `coerce` (see https://github.com/purescript/purescript-newtype/pull/22). This is only a breaking change for non derived instances because the `Newtype` class has no members anymore and can now only be implemented for representationally equal types (those satisfying the new superclass constraint).
+
+For example the instance for `newtype Additive a = Additive a` no longer implements `unwrap` and `wrap`:
+
+```diff
++instance newtypeAdditive :: Newtype (Additive a) a
+-instance newtypeAdditive :: Newtype (Additive a) a where
+-  wrap = Additive
+-  unwrap (Additive a) = a
+```
+
+Derived instances don't require any modifications.
+
+* Reform handling of quote characters in raw strings (#3961, @rhendric)
+
+Quotes behaved rather unexpectedly in various edge cases inside raw strings. This clears things up by enforcing the following specification:
+
+```
+'"""' '"'{0,2} ([^"]+ '"'{1,2})* [^"]* '"""'
+```
+
+Meaning that raw strings can contain up to two successive quotes, any number of times, but three successive quotes are not allowed inside.
+
+* Unsupport bare negative literals as equational binders (#3956, @rhendric)
+
+It used to be possible to match on negative literals, such as `-1`, but this prevented parsing matches on constructors aliased to `-`. The compiler will reject matches on _bare_ negative literals, but they can still be matched by wrapping them in parentheses.
+
+* Forbid partial data constructors exports (#3872, @kl0tl)
+
+Exporting only some of the constructors of a type meant that changes internal to a module, such as adding or removing an unexported constructor, could cause unexhaustive pattern match errors in downstream code. Partial explicit export lists will have to be completed with the missing constructors or replaced by implicit export lists.
+
+* Print compile errors to stdout, progress messages to stderr (#3839, @JordanMartinez)
+
+The compiler used to print errors to stderr, which is especially annoying when piping its output to another program with the `--json-errors` flag. Stderr is really for diagnostics about the execution of a program, whereas its expected output ought to be printed to stdout.
```

Comment:
I think this is a little opinionated, and it might be better focus a bit more on the "what" rather than the "why". When you say "its expected output ought to be printed to stdout", you haven't actually said where it _is_ printed, only where you think it should be printed. How about:
```suggestion
Compiler errors and warnings arising from your code are now printed to stdout rather than stderr, and progress messages such as "Compiling Data.Array" are now printed to stderr rather than stdout. Warnings and errors arising from incorrect use of the CLI, such as specifying input files which don't exist or specifying globs which don't match any files, are still printed to stderr (as they were before). This change is useful when using the `--json-errors` flag, since  you can now pipe JSON errors into other programs without having to perform awkward gymnastics such as `2>&1`.
```